python/ray/_private/ray_logging/logging_config.py

```diff
@@ -1,13 +1,87 @@
-from ray._private.ray_logging.constants import LOG_MODE_DICT
+from abc import ABC, abstractmethod
+from typing import Set
+
+from ray._private.ray_logging import default_impl
 from ray.util.annotations import PublicAPI
 
 from dataclasses import dataclass
 
 
+class DictConfigProvider(ABC):
+    @abstractmethod
+    def get_supported_encodings(self) -> Set[str]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_dict_config(self, encoding: str, log_level: str) -> dict:
+        raise NotImplementedError
+
+
+class DefaultDictConfigProvider(DictConfigProvider):
+    def __init__(self):
+        self._dict_configs = {
+            "TEXT": lambda log_level: {
+                "version": 1,
+                "disable_existing_loggers": False,
+                "formatters": {
+                    "text": {
+                        "()": "ray._private.ray_logging.formatters.TextFormatter",
```

Comment:
wondering if you want to do:

`"()": f"{TextFormatter.__module__}{TextFormatter.__qualname__}"`

this way the class name is synced with the code